tests/pass-dep/concurrency/windows_condvar_shared.rs

```diff
@@ -2,42 +2,29 @@
 // We are making scheduler assumptions here.
 //@compile-flags: -Zmiri-preemption-rate=0
 
-use std::ffi::c_void;
-use std::ptr::null_mut;
+use std::mem::MaybeUninit;
 use std::thread;
 
+use windows_sys::Win32::System::Threading::{
+    AcquireSRWLockExclusive, AcquireSRWLockShared, ReleaseSRWLockExclusive, ReleaseSRWLockShared,
+    SleepConditionVariableSRW, WakeAllConditionVariable, CONDITION_VARIABLE_LOCKMODE_SHARED,
+    INFINITE,
+};
+
 #[derive(Copy, Clone)]
 struct SendPtr<T>(*mut T);
 
 unsafe impl<T> Send for SendPtr<T> {}
 
-extern "system" {
-    fn SleepConditionVariableSRW(
-        condvar: *mut *mut c_void,
-        lock: *mut *mut c_void,
-        timeout: u32,
-        flags: u32,
-    ) -> i32;
-    fn WakeAllConditionVariable(condvar: *mut *mut c_void);
-
-    fn AcquireSRWLockExclusive(lock: *mut *mut c_void);
-    fn AcquireSRWLockShared(lock: *mut *mut c_void);
-    fn ReleaseSRWLockExclusive(lock: *mut *mut c_void);
-    fn ReleaseSRWLockShared(lock: *mut *mut c_void);
-}
-
-const CONDITION_VARIABLE_LOCKMODE_SHARED: u32 = 1;
-const INFINITE: u32 = u32::MAX;
-
 /// threads should be able to reacquire the lock while it is locked by multiple other threads in shared mode
 fn all_shared() {
     println!("all_shared");
 
-    let mut lock = null_mut();
-    let mut condvar = null_mut();
```

Comment:
You also changed this from null ptrs to pointers to some actual memory. Why that?

Reply:
They're pointer sized structures. In the original tests, I just made them pointers for simplicity. `windows-sys` uses separate types for them, so I thought this was the simplest way to zero initialize them. `mem::zeroed` would also work here, but requires more unsafe blocks.

Reply:
Oh and `SRWLOCK/CONDITION_VARIABLE/INIT_ONCE { Ptr: null_mut() }` also works too, but is bigger looking and requires more imports.

Reply:
Ah I see, the tests previously used imprecise types.

According to [these docs](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleepconditionvariablesrw) what we do here is anyway questionable...
"This variable must be initialized using the InitializeConditionVariable function."

I guess Miri doesn't currently check that the variable is initialized at all. But that's a pre-existing issue, nothing to do with this PR.

Reply:
There are constants to statically initialize these types mentioned in [these docs](https://learn.microsoft.com/en-us/windows/win32/sync/slim-reader-writer--srw--locks)
> The caller must allocate an SRWLOCK structure and initialize it by either calling [InitializeSRWLock](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initializesrwlock) (to initialize the structure dynamically) or assign the constant SRWLOCK_INIT to the structure variable (to initialize the structure statically).

`SRWLOCK_INIT`, `CONDITION_VARIABLE_INIT` and `INIT_ONCE_STATIC_INIT` are all equal to `{0}`. They aren't in `windows-sys` though.